### PR TITLE
Support for ATMega EXT & INT external interrupts.

### PR DESCRIPTION
**Pull request by igor-stoppa**
This PR adds unified support for the external interrupt sources in ATMega chips.
Tested on ATMega328p (Arduino mini pro) with optical end-stop detector as source of interrupts.
The PR also contains an example under the testhal folder.

### CODE REVIEW

File: testhal/AVR/EXT/mcuconf.h [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+/*
+ * EXT driver system settings.
+ */
+#define AVR_EXT_USE_EXT1                    TRUE
+
+/*
+ * PC driver system settings.
+ */
+#define AVR_PC_USE_PC1                      TRUE
```
Comment (utzig) [inline testhal/AVR/EXT/mcuconf.h]: A better name for this would be AVR_EXT_USE_PCINT1

Comment (utzig) [inline testhal/AVR/EXT/mcuconf.h]: Although a more correct naming would be AVR_EXT_USE_EXT2 (to differentiate, but sound strange!)

File: testhal/AVR/EXT/halconf.h [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+ */
+#if !defined(HAL_USE_TM) || defined(__DOXYGEN__)
+#define HAL_USE_TM                  FALSE
+#endif
+
+/**
+ * @brief   Enables the PC subsystem.
+ */
+#if !defined(HAL_USE_PC) || defined(__DOXYGEN__)
+#define HAL_USE_PC                  FALSE
```
Comment (utzig) [inline testhal/AVR/EXT/halconf.h]: This should be removed.

**Comment by utzig** [comment]
Sorry but I will not be able to finish reviewing today. Looks mostly good from what I could see. But please, please don't close this pull request again to open another one! You can always update a pull request with new commits if that turns out to be required.

File: testhal/AVR/EXT/mcuconf.h (hunk 1 above)
Comment (igor-stoppa) [inline testhal/AVR/EXT/mcuconf.h]: Oh, that is just an error: a leftover from the previous implementation with 2 separate drivers.
Thanks for spotting it.

File: testhal/AVR/EXT/halconf.h (hunk 2 above)
Comment (igor-stoppa) [inline testhal/AVR/EXT/halconf.h]: Ditto.

**Comment by igor-stoppa** [comment]
Apologies for the pull request mess. I'm more used to mailing lists, still learning the ropes with GitHUB.
I'll avoid repeating this.

**Comment by igor-stoppa** [comment]
I've updated the pull request based on the 2 comments received.

**Comment by barthess** [comment]
Sorry for this merge. I reverted it back. Any way, this is read only mirror. How you plan to push your improvements in main SVN repo?

**Comment by igor-stoppa** [comment]
@barthess : @utzig said I could share the code with him for review, so I created a PR here.
What is the correct target for a PR?
I've been using this code for more than a month and it works for me ... considering that it only adds functionality, it doesn't seem to risky to merge it. I'd just like to get done with this and move on to the next driver I want to send a PR for. The only 2 comments I got have been addressed already - I was actually waiting for more comments, but I got tired and fixed these 2, in the meanwhile.

**Comment by barthess** [comment]
Suppose utzig wants to create patch from your PR and merge it into SVN repository where he has write access. Github is more convenient for code review. Did you read this article? http://wiki.chibios.org/dokuwiki/doku.php?id=chibios:community:guides:community_overlay

**Comment by igor-stoppa** [comment]
@barthess : no, I had not noticed, thanks.
But that doesn't seem to work for certain things. At least I'd need an explanation.

This specific PR would probably go in fine, but in the next I'd like to rename the directory containing the PWM example to something like PWM-ARDUINOMEGA and then create a further PWM-ARDUINOUNO example. Because the example (and others as well) is only for ArduinoMega and won't work on Uno.

From the Contrib project, I cannot rename/modify a file that is in the Core project.

To me the Contrib part seems fine when one wants to add completely new code (say a port to a new architecture or a new device driver), but it's not usable when trying to patch the Core code.
Should I create a bug for the core Code and attach my patch?

**Comment by barthess** [comment]
You can contribute your changes via forum topic in relevant thread or via bugtracker on sourceforge. First method is preferred.

Platform.mk changes is generally not a problem because you can create new one in Contrib and include them into project's makefile. See for example https://github.com/ChibiOS/ChibiOS-Contrib/blob/master/testhal/STM32/STM32F4xx/onewire/Makefile#L89

**Pull request merged**